adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/V3.kt

```diff
@@ -46,27 +47,31 @@ problems as an issue in the <a href=\"https://github.com/AdoptOpenJDK/openjdk-ap
 )
 @ApplicationScoped
 @ApplicationPath("/")
-class V3 : Application {
+class V3 : Application() {
 
     companion object {
         val ENABLE_PERIODIC_UPDATES: String = "enablePeriodicUpdates"
     }
 
-    private val apiDataStore: APIDataStore
+    private lateinit var apiDataStore: APIDataStore
     private val resourceClasses: Set<Class<out Any>>
     private val cors: Set<Any>
 
     @Inject
-    constructor(apiDataStore: APIDataStore) {
```

Comment:
constructor injection didn't work anymore, too.
jax-rs seems also not to require it, changed to setter injection and `@PostConstruct` initialization.

Reply:
I think this is probably a bug in ResteasyServerCommonProcessor, it seemed for me that even the presence of an empty constructor on the Application fixed this, i.e adding:

```
constructor() {}
```

To V3, and it seemed to work correctly. The other constructor with Inject was actually used.

I am a bit 50/50 on this as the empty constructor still means we need to have apiDataStore as lateinit, but I think I would prefer the constructor injection.

Reply:
Thank you for your thoughts, in general I also prefer constructor injection.
I did found discussions about this in Quarkus issues as well as on stackoverflow, the essence was, that one cannot relay that constructor injection works in the jax-rs Application class.

I try, if the constructor injection works if there's an additional empty constructor - if so, that's the winner.